templates/al2023/template.json

```diff
@@ -213,6 +213,10 @@
       "remote_folder": "{{ user `remote_folder`}}",
       "script": "{{template_dir}}/provisioners/install-nodeadm.sh",
       "environment_vars": [
+        "AWS_ACCESS_KEY_ID={{user `aws_access_key_id`}}",
```

Comment:
we should use the instance profile created by Packer and give it temporary permissions via `temporary_iam_instance_profile_policy_document`. this will be cleaner for those who are creating custom AMIs as well